### PR TITLE
refactor into file per action

### PR DESCRIPTION
**Pull request by sparrell**

### CODE REVIEW

File: apps/ocas/src/act_allow.erl [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            lager:debug( "allow_server got keepalive" ),
+            From!{keepalive_received, allow_server},
+            allow_server(State);
+        %% stop server - note it doesnt loop
+        stop_server ->
+            lager:debug( "allow_server stopping" ),
+            stopping;
+        %% handle unaccounted for message
+        _ ->
+            lager:debug( "allow_server server got something not accounted for" ),
```
Comment (elvisci) [inline apps/ocas/src/act_allow.erl]: Line 62 is too long:             lager:debug( "allow_server server got something not accounted for" ),.

File: apps/ocas/src/openc2_handler.erl [hunk 2]
```diff
@@ -135,28 +128,29 @@ has_action(true, Req, State ) ->
     JsonMap = maps:get(json_map, State2),
     ActionBin = maps:get( <<"action">>, JsonMap ),
     lager:info("action bintext: ~p", [ActionBin] ),
-    { ActionValid, ActionValue } = actions:is_valid_action(ActionBin),
+    { ActionValid, {ActionModule, ActionFunction} } = actions:is_valid_action(ActionBin),
```
Comment (gadgetci) [inline apps/ocas/src/openc2_handler.erl]: According to **Elvis**:
> Line 131 is too long:     { ActionValid, {ActionModule, ActionFunction} } = actions:is_valid_action(ActionBin),.

File: apps/ocas/src/actions.erl [hunk 3]
```diff
@@ -86,56 +49,57 @@ is_valid_action(Action) ->
...
     ActionValid = maps:is_key(Action, ValidActions),
-    ActionValue = maps:get(Action, ValidActions, undefined),
+    DefaultAction = { undefined, undefined },
+    ActionValue = maps:get(Action, ValidActions, DefaultAction),
     { ActionValid, ActionValue }.
 
-spawn_action( ActionServer, Json ) ->
-    lager:info( "Got to spawn_action for ~p: ", [ ActionServer ] ),
+spawn_action( ActionModule, ActionServer, Json ) ->
+    lager:info( "Got to spawn_action for ~p:~p ", [ ActionModule, ActionServer ] ),
```
Comment (gadgetci) [inline apps/ocas/src/actions.erl]: According to **Elvis**:
> Line 99 is too long:     lager:info( "Got to spawn_action for ~p:~p ", [ ActionModule, ActionServer ] ),.

File: apps/ocas/src/openc2_handler.erl [hunk 4]
```diff
@@ -135,28 +128,29 @@ has_action(true, Req, State ) ->
...
     State3 = maps:put(action_valid, ActionValid, State2),
-    State4 = maps:put(action, ActionValue, State3),
+    State4 = maps:put(action_module, ActionModule, State3),
+    State5 = maps:put(action_function, ActionFunction, State4),
 
     %% check if action is valid
-    check_valid_action( ActionValid, ActionValue, Req, State4 ).
+    check_valid_action( ActionValid, ActionModule, ActionFunction, Req, State5 ).
```
Comment (elvisci) [inline apps/ocas/src/openc2_handler.erl]: Line 137 is too long:     check_valid_action( ActionValid, ActionModule, ActionFunction, Req, State5 )..

Comment (gadgetci) [inline apps/ocas/src/openc2_handler.erl]: According to **Elvis**:
> Line 137 is too long:     check_valid_action( ActionValid, ActionModule, ActionFunction, Req, State5 )..

File: apps/ocas/src/act_alert.erl [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            lager:debug( "alert_server got keepalive" ),
+            From!{keepalive_received, alert_server},
+            alert_server(State);
+        %% stop server - note it doesnt loop
+        stop_server ->
+            lager:debug( "alert_server stopping" ),
+            stopping;
+        %% handle unaccounted for message
+        _ ->
+            lager:debug( "alert_server server got something not accounted for" ),
```
Comment (gadgetci) [inline apps/ocas/src/act_alert.erl]: According to **Elvis**:
> Line 62 is too long:             lager:debug( "alert_server server got something not accounted for" ),.

File: apps/ocas/src/openc2_handler.erl (hunk 2 above)
Comment (elvisci) [inline apps/ocas/src/openc2_handler.erl]: Line 131 is too long:     { ActionValid, {ActionModule, ActionFunction} } = actions:is_valid_action(ActionBin),.

File: apps/ocas/src/act_allow.erl (hunk 1 above)
Comment (gadgetci) [inline apps/ocas/src/act_allow.erl]: According to **Elvis**:
> Line 62 is too long:             lager:debug( "allow_server server got something not accounted for" ),.

File: apps/ocas/src/actions.erl (hunk 3 above)
Comment (elvisci) [inline apps/ocas/src/actions.erl]: Line 99 is too long:     lager:info( "Got to spawn_action for ~p:~p ", [ ActionModule, ActionServer ] ),.

File: apps/ocas/src/act_mitigate.erl [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            lager:debug( "mitigate_server got keepalive" ),
+            From!{keepalive_received, mitigate_server},
+            mitigate_server(State);
+        %% stop server - note it doesnt loop
+        stop_server ->
+            lager:debug( "mitigate_server stopping" ),
+            stopping;
+        %% handle unaccounted for message
+        _ ->
+            lager:debug( "mitigate_server server got something not accounted for" ),
```
Comment (gadgetci) [inline apps/ocas/src/act_mitigate.erl]: According to **Elvis**:
> Line 62 is too long:             lager:debug( "mitigate_server server got something not accounted for" ),.

**Pull request merged**